numpy/lib/npyio.py

```diff
@@ -253,6 +255,138 @@ def iterkeys(self):
     def __contains__(self, key):
         return self.files.__contains__(key)
 
+class IncrementalWriter:
+    """
+    Class for the incremental writing of binary ``.npy`` files.
+    Multiple calls to ``save()`` append data to one binary ``.npy``
+    file resulting in one concatenated array saved. Saved arrays must
+    have exactly the same dtype and have compatible shapes. Shapes can
+    differ in the 1st dimension only which will be added up in the file.
+    The file is written in C order.
+
+    Parameters
+    ----------
+    fp : file-like object or string
+        Name of the file to open or a file-like object supporting
+        ``tell()``, ``seek()`` and ``write()``. In the latter case,
+        data will be written from current position.
+    flush : bool, optional
+        Calls ``flush()`` at the end of each ``save()``. Default:
+        False
+    hdrupdate : bool, optional
+        If True, updates header after each call to ``save()`` by seeking back
+        to the beginning (starting position) of the file, rewrites
+        header then seeks to the current position again.  Otherwise header
+        will be updated once at ``close()``.  Default: True.
+    version : tuple or None, optional
+       The .npy file version to use or to detect automatically at
+       first ``save()`` call if None.
+
+    See Also
+    --------
+    load
+    """
+    #TODO: implement header reading to open for append and for partial
+    #reading.
+
+    EXTRAPAD=32 # We allow 32 bytes to grow in the shape description
+
+    def __init__(self, fp, flush=False, hdrupdate=True,version=None):
+        if isfileobj(fp):
+            self.fp = fp
+            self.hfilepos = fp.tell()
+        else:
+            self.fp = open(fp, 'wb')
+            self.hfilepos = 0
+        self.flush = flush
+        self.hdisklen = 0
+        self.hdrupdate = hdrupdate
+        self.version = version
```

Comment:
These should probably be private attributes (like `self._fp`, not `self.fp`). Also it would be nice to use real words -- `hfilepos` and `hdisklen` don't mean much to me :-)